src/Refactoring-Transformations/RBPullUpMethodTransformation.class.st

```diff
@@ -291,26 +328,25 @@ RBPullUpMethodTransformation >> removeDuplicateMethods [
 
 { #category : 'executing' }
 RBPullUpMethodTransformation >> removeDuplicatesOf: aSelector [
-	^ RBCustomTransformation
-		model: self model
-		with: [ :rbModel |
-			| tree |
-			tree := targetSuperclass parseTreeForSelector: aSelector.
-			targetSuperclass allSubclasses collect:
-			[:aClass |
-			((aClass directlyDefinesMethod: aSelector) and:
-					[(tree equalTo: (aClass parseTreeForSelector: aSelector) exceptForVariables: #())
-						and: [(aClass superclass whoDefinesMethod: aSelector) == targetSuperclass]])
-				ifTrue:
-					[removeDuplicates
-						ifFalse:
-							[removeDuplicates := self refactoringConfirmWarning:
-							'Do you want to remove duplicate subclass methods?'].
-					removeDuplicates ifTrue:[
-						self generateChangesFor: (RBRemoveMethodTransformation
-							model: self model
-							selector: aClass
-							from: aSelector) ]]]]
+
+	^ RBCustomTransformation model: self model with: [ :rbModel |
+		  | tree |
+		  tree := targetSuperclass parseTreeForSelector: aSelector.
+		  targetSuperclass allSubclasses collect: [ :aClass |
+			  ((aClass directlyDefinesMethod: aSelector) and: [
+				   (tree
+					    equalTo: (aClass parseTreeForSelector: aSelector)
+					    exceptForVariables: #(  )) and: [
+					   (aClass superclass whichClassIncludesSelector: aSelector)
+					   == targetSuperclass ] ]) ifTrue: [
+				  removeDuplicates ifFalse: [
+					  removeDuplicates := self refactoringConfirmWarning:
+						                      'Do you want to remove duplicate subclass methods?' ].
+				  removeDuplicates ifTrue: [
```

Comment:
We want to remove all these warnings to driver from both transformation and refactoring. We can introduce flags for scripting purpose so user can configure it and we will set sane defaults.

Reply:
I modified [issue#17158](https://github.com/pharo-project/pharo/issues/17158) to handle this too